flight_club/sessions/views.py

```diff
@@ -19,65 +19,253 @@
+        )
+
+        if not session_valid:
+            flash(error)
+            return render_template("sessions/add_session.html")
         else:
-            # Check Session isn't duplicate
             session_id = request.form["session_id"]
-            if db_func.check_if_session_exists(session_id):
-                error = "This is a duplicate session"
-            else:
-                # Add the session
-                date = request.form["date"]
-                db.session.add(Session(id=session_id, date=date))
-                db.session.commit()
-
-                # Extract Number of Beers
-                num_beers = int(request.form["beers"])
-
-                # Loop Through Form
-                # TODO (dan) there is a way for this form to submit without there being beers
-                # should probably fix the front end..
-                for i in range(num_beers):
-                    beer_name = request.form["beer_name{}".format(i)]
-                    brewery = request.form["brewery{}".format(i)]
-                    style = request.form["style{}".format(i)]
-                    votes = request.form["votes{}".format(i)]
-                    win = request.form["win{}".format(i)]
-                    username = request.form["username{}".format(i)]
-
-                    db.session.add(
-                        Beer(
-                            beer_name=beer_name,
-                            brewery=brewery,
-                            style=style,
-                            votes=votes,
-                            win=win,
-                            username=username,
-                            session_id=session_id,
-                        )
+            # Construct and add the session
+            db.session.add(new_session)
+
+            # Helper variables for beer loop
+            still_beers = True
+            beer_list = []
+            i = 0
+
+            while still_beers:
+                try:
+                    beer_name = request.form["beer_{}".format(i)]
+                    beer_abv = float(request.form["beer_abv_{}".format(i)])
+                    brewery = request.form["brewery_{}".format(i)]
+                    style = request.form["style_{}".format(i)]
+                    votes = request.form["votes_{}".format(i)]
+                    win = request.form["win_{}".format(i)]
+                    username = request.form["username_{}".format(i)]
+
+                    beer_valid, error, new_beer = beer_input_validator(
+                        beer_name,
+                        beer_abv,
+                        brewery,
+                        style,
+                        votes,
+                        win,
+                        username,
+                        session_id,
                     )
+                    if beer_valid:
+                        beer_list.append(new_beer)
+                        i += 1
```

Comment:
Why are we tracking i here?  It's a weak variable name and more robustly we should just be checking `len(beer_list)` anywhere we're using i

Reply:
Oh yeah definitely can use len(beer_list) actually. Forgot to go back and do that. At first I didn't have a beer list.